src/ephysatlas/feature_computation.py

```diff
@@ -175,86 +187,110 @@ def online_feature_computation(
+                f"Requested time range ({t0} to {t0 + duration}) exceeds AP data duration ({max_time_ap})"
+            )
+        
+        # Calculate start indices for data access
+        n0_ap = int(sr_ap.fs * t0)
 
-    # Calculate the next fast length for the LF data to optimize FFT operations
-    ns_lf = scipy.fft.next_fast_len(int(sr_lf.fs * duration), real=True)
+        # Verify channel indices are valid
+        n_channels_ap = sr_ap.nc - sr_ap.nsync
 
-    # Check if requested time range is within bounds of available data
-    max_time_ap = sr_ap.ns / sr_ap.fs
-    max_time_lf = sr_lf.ns / sr_lf.fs
+        # Validate channel counts
+        if n_channels_ap <= 0:
+            raise ValueError(
+                f"Invalid number of channels: AP={n_channels_ap}"
+            )
 
-    # Validate start time is non-negative
-    if t0 < 0:
-        raise ValueError(f"Start time t0 ({t0}) cannot be negative")
-    # Validate AP data duration
-    if t0 + duration > max_time_ap:
-        raise ValueError(
-            f"Requested time range ({t0} to {t0 + duration}) exceeds AP data duration ({max_time_ap})"
-        )
-    # Validate LF data duration
-    if t0 + duration > max_time_lf:
-        raise ValueError(
-            f"Requested time range ({t0} to {t0 + duration}) exceeds LF data duration ({max_time_lf})"
-        )
+        # Load AP data, ignoring sync pulse columns
+        try:
+            raw_ap = sr_ap[slice(n0_ap, n0_ap + ns_ap), :n_channels_ap].T
+        except IndexError as e:
+            raise IndexError(
+                f"Failed to access AP data: {str(e)}. Check if time range or channel count is valid."
+            )
 
-    # Calculate start indices for data access
-    n0_ap = int(sr_ap.fs * t0)
-    n0_lf = int(sr_lf.fs * t0 + 3)  # Add 3 to account for LF latency
+    else:
+        raw_ap = None
+    
+    if sr_lf is not None:
+        # Calculate the next fast length for the LF data to optimize FFT operations
+        ns_lf = scipy.fft.next_fast_len(int(sr_lf.fs * duration), real=True)
+        max_time_lf = sr_lf.ns / sr_lf.fs
+        # Validate AP data duration
 
-    # Verify channel indices are valid
-    n_channels_ap = sr_ap.nc - sr_ap.nsync
-    n_channels_lf = sr_lf.nc - sr_lf.nsync
+        # Validate LF data duration
+        if t0 + duration > max_time_lf:
+            raise ValueError(
+                f"Requested time range ({t0} to {t0 + duration}) exceeds LF data duration ({max_time_lf})"
+            )
+        
 
-    # Validate channel counts
-    if n_channels_ap <= 0 or n_channels_lf <= 0:
-        raise ValueError(
-            f"Invalid number of channels: AP={n_channels_ap}, LF={n_channels_lf}"
-        )
+        # Calculate start indices for data access
+        n0_lf = int(sr_lf.fs * t0 + 3)  # Add 3 to account for LF latency
+
+        # Verify channel indices are valid
+        n_channels_lf = sr_lf.nc - sr_lf.nsync
+
+        # Validate channel counts
```

Comment:
I think this validation happens in spikeglx Reader already. It looks like AI bloat TBH !